app/views/apipie_dsl/apipie_dsls/help.html.erb

```diff
@@ -1,58 +1,340 @@
+                           apipie_erb_wrap('result"'),
+                       "host1.example.com\npositive") %>
+
+<p><%= _('Note that the else branch is optional. Also multiple conditions can be specified using elsif, but that is usually better written using a case/when construct. See the example below') %></p>
+
+<%= apipie_dsl_example(apipie_erb_wrap("case input('Order')", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("when 'Name'", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("  result = 'Name'", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("when 'Name and Age'", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("  result = [ 'Name', 'Age' ]", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("else", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("  result = 'Age'", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("end", mode: :silent, open_trim: true, close_trim: true) + "\n" +
+                           apipie_erb_wrap("report_render order: result", close_trim: true)) %>
+
+<p><%= _('Note that the else branch is optional and can be omitted.') %></p>
+
+<h3 id="iterating"><%= _('Iterating over Arrays') %></h3>
+
+<p><%= _('Sometimes it is necessary to iterate over arrays in order to retrieve the information. This is typically required in report templates when loaders are used, however the same technique is useful elsewhere too.') %></p>
+
+<p><%= _('For example, this can be useful to configure all the Host NICs.') %></p>
+
+<%= apipie_dsl_example(apipie_erb_wrap("@host.interfaces", close_trim: true), "#<Nic::Base::ActiveRecord_Associations_CollectionProxy:0x00007f734036fbe0>") %>
+
+<p><%= _('As you can see in the example above, the <code>interfaces</code> method on Host object returns a new object of type Collection. For simplicity, think of it as an array of all host interfaces. To process every item in this array, a method <code>each</code> is used.').html_safe %></p>
+
+<%= apipie_dsl_example(apipie_erb_wrap("@host.interfaces.each do |iface|", open_trim: true, mode: :silent) + "\n" +
+                           apipie_erb_wrap("  iface.ip") + "\n" +
+                           apipie_erb_wrap("end", open_trim: true, close_trim: true, mode: :silent),
+                       "192.168.122.144\n10.0.0.1\n") %>
+
+<p><%= _('This is the simple example of iterating over array of two interfaces and only printing the IP address of each. In reality, the interface object can be used to fetch more information, as shown in the following example. To see all available methods, see the documentation for Nic::Managed object.') %></p>
```

Comment:
maybe you can give a link here to Nic::Managed object?

Reply:
ok, found a decent way to generate the link - `link_to('Nic::Managed', apipie_dsl_apipie_dsl_path(version: ApipieDSL.configuration.default_version, section: 'all', class: 'Nic::Base')`, updating